upgrade: Upgrade Python version form 3.9 to 3.10 and refactor code base.

Swangeon opened the pull request:
Add version changes to of both python and pipenv to README.md. 
Add`bandit` module as its also needed but not in Pipfile.

aristofanischionis commented:
Hi Swangeon, could you please remove the tool-generated files as we discussed? Also, could you perform the soft syntax updates? To clean up the code. Then I can review all the changes together.
Best

Swangeon commented:
Will do! This PR will probably be up for some time longer but I will get those done.

Swangeon commented:
Everything to my knowledge has been changed to conform more with PEP8. There are a couple of things that I didn't decide on changing to conform to PEP8 like long strings or long paths as splitting those up to get to the 79 line length made the code harder to read in my opinion. Type hints should be there for most if not all variables and created some type aliases to refer to certain data structures that were nested and/or had a lot of types that could be returned. Added the runtime generated output files to the .gitignore (Will do the reports directory later as I am doing the revamp of the reporting system). Fixed minor things like the output of the port scanner as `os` became `osmatch` and `hostnames` became `hostname` and how the version check for Conpot is done.

aristofanischionis commented on honeyscanner/Pipfile:
```diff
@@ -4,16 +4,16 @@ verify_ssl = true
 [requires]
-python_version = "3.9"
-python_full_version = "3.9.12"
+python_version = "3.10"
```
let's upgrade this to the latest python version 3.12

aristofanischionis commented on honeyscanner/Pipfile:
```diff
@@ -4,16 +4,16 @@ verify_ssl = true
-python_version = "3.9"
-python_full_version = "3.9.12"
+python_version = "3.10"
+python_full_version = "3.10.14"
```
and this 3.12.4

aristofanischionis reviewed (approved):
Tested and works well on my MacOS host system
